Coloma/Coloma.cs

```diff
@@ -152,66 +156,41 @@ private static void SortandFix(List<ColomaEvent> eventlist, bool setuplog)
             {
-                if (evt.Logname == "Setup")
-                {
-                    r = evt.Revision;
-                }
-                else
-                {
-                    evt.Revision = r;
-                }
+                evt.Revision = r;
             }
         }
 
-        private static bool AddSetupLogToList(List<KBRevision> kbrlist, List<ColomaEvent> list)
+        private static uint? GetRevisionFromSetupLog(List<KBRevision> kbrlist)
         {
-            // this retrieves the build.revision and branch for the current client
-            WindowsVersion.WindowsVersionInfo wvi = new WindowsVersion.WindowsVersionInfo();
-            WindowsVersion.GetWindowsBuildandRevision(wvi);
-            uint revision = wvi.revision;
-            bool setuplog = false;
-
-            EventLogQuery query = new EventLogQuery("Setup", PathType.LogName);
-            query.ReverseDirection = false;
+            EventLogQuery query = new EventLogQuery("Setup", PathType.LogName) {ReverseDirection = false};
             EventLogReader reader = new EventLogReader(query);
 
             EventRecord entry;
             while ((entry = reader.ReadEvent()) != null)
             {
-                if ((entry.Level == (byte)StandardEventLevel.Critical) ||
-                    (entry.Level == (byte)StandardEventLevel.Error) ||
-                    (entry.Level == (byte)StandardEventLevel.Warning) ||
-                    (entry.Id == 2))
+                if (entry.Id != 2) continue;
+
+                // this is a KB installed message, figure out which KB it is and update the revision of that entry
+                string msg = CleanUpMessage(entry.FormatDescription());
+                string kb = "KB";
+                int i = msg.IndexOf(kb);
+
+                if (-1 != i)
                 {
-                    string msg = CleanUpMessage(entry.FormatDescription());
+                    // we found the kb article
+                    kb = msg.Substring(i, 9);
 
-                    if (entry.Id == 2)
+                    foreach (KBRevision rev in kbrlist)
                     {
-                        // this is a KB installed message, figure out which KB it is and update the revision of that entry
-                        string kb = "KB";
-                        int i = msg.IndexOf(kb);
-                        setuplog = true;
-
-                        if (-1 != i)
+                        if (rev.Kb == kb)
                         {
-                            // we found the kb article
-                            kb = msg.Substring(i, 9);
-
-                            foreach (KBRevision rev in kbrlist)
-                            {
-                                if (rev.Kb == kb)
-                                {
-                                    revision = rev.Revision;
-                                }
-                            }
+                            return rev.Revision;
```

Comment:
If the logs collected span multiple revisions, this will only return the earliest revision number rather than the most recent. I don't understand why this would be desirable.